numba/core/dispatcher.py

```diff
@@ -995,6 +1056,20 @@ def get_function_type(self):
             cres = tuple(self.overloads.values())[0]
             return types.FunctionType(cres.signature)
 
+    def _call_tls_target(self, args, kwargs):
+        # Check TLS target configuration
+        tc = TargetConfig()
+        retarget = tc.get()
+        # Check cache
+        siblings = self._siblings
+        disp = siblings.load_cache(retarget)
+        if disp is None:
+            # make new
+            disp = retarget(self)
+            siblings.save_cache(retarget, disp)
```

Comment:
In out-of-band discussion with DPPY developers, we want make the cache more flexible and easier to be customized.